### ISSUE TITLE
Visual Studio 2022 crashes when i try to open "manage python packages"

### ISSUE THREAD

**Issue by vsfeedback** [microsoft/PTVS#6944]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Visual-Studio-2022-crashes-when-i-try-to/1698469)._

---
[severity:I'm unable to use this version]
Hi! 

When i right-click on the python environment and press "Manage Python Packages" visual studio crashes and reboots. 
I have tried upgrading, i´m now on 17.1.1 (2022 Community 64-bit).

I´m on windows 10, just over 2 GHz, 32 GB ram, Gforce 1050.

The nuget package manager is 6.1.0
i have tried with python 3.9 and 3.7.9

I don´t really know what more to say, i have attached a logfile screenshot (at least i think so).

---
#### Feedback Bot on 3/22/2022, 06:25 AM: 

(private comment, text removed)
#### Martin Lith on 3/22/2022, 08:46 AM: 

(private comment, text removed)
#### Martin Lith on 3/22/2022, 10:47 AM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by heathermrauch** [comment]
I am experiencing the same issue. Additionally, the conda environment I was using with VS 2019 was deleted including all of my variables. This version of the software is trash! Please try again!

**Comment by asm128** [comment]
Same problem. Also it seems there are some invalid environments it tries to detect before the crash.

**Comment by judej** [comment]
@heathermrauch, Sorry for the issues you are seeing.  Unfortunately, we have been unable to reproduce this. Could you please try installing the latest Python from python.org. There have been some issues with `pip` and want to eliminate that possibility.

@asm128, what are the invalid environments errors that you are seeing?

**Comment by heathermrauch** [comment]
@judej, I uninstalled VS 2022 and reverted to 2019. Because of this issue, I had to completely rebuild my environment from scratch and was unable to do so using the latest version of python as some of the libraries on which my applications depend are not compatible with the latest release. Because of all the rework my first attempt at this took, I have no intention of reinstalling the latest version of Visual Studio or attempting to reproduce the issue. I will more likely switch which IDE I use in the future since I have already been experiencing issues with Visual Studio crashing for no apparent reason. I can guarantee I am not the only developer who is doing so because of the increasingly unstable releases of this product.

**Comment by asm128** [comment]
> @heathermrauch, Sorry for the issues you are seeing. Unfortunately, we have been unable to reproduce this. Could you please try installing the latest Python from python.org. There have been some issues with `pip` and want to eliminate that possibility.
> 
> @asm128, what are the invalid environments errors that you are seeing?

I have this problem in another windows 10 just installed with miniconda 3 also present and probably Python 10 too. Nothing else.

**Comment by asm128** [comment]
The conda env i created only has opencv-python installed

**Comment by asm128** [comment]
> 

You should do like me and write code that doesn't get old by relying less on experimental and or third-party stuff

**Comment by heathermrauch** [comment]
> The conda env i created only has opencv-python installed
> 
> @heathermrauch You should do like me and write code that doesn't get old by relying less on experimental and or third-party stuff

@asm128, thanks for the advice. Unfortunately that is not an option since the purpose of my code is to integrate with the API for the "third party stuff".

**Comment by asm128** [comment]
> > The conda env i created only has opencv-python installed
> > @heathermrauch You should do like me and write code that doesn't get old by relying less on experimental and or third-party stuff
> 
> @asm128, thanks for the advice. Unfortunately that is not an option since the purpose of my code is to integrate with the API for the "third party stuff".

You can always upgrade what's outdated to the latest version of everyhing and implement what's missing if there's something that got too old or is too difficult to maintain. Why keeping supporting code that doesn't work on the latest release of the language or whatever? I brought to date large C++ projects that were written for Visual C++ 6 without any real difficulties (just they didn't even had to declare the type of the varirable and it assumed int for unknown vars.

**Comment by heathermrauch** [comment]
> > > The conda env i created only has opencv-python installed
> > > @heathermrauch You should do like me and write code that doesn't get old by relying less on experimental and or third-party stuff
> > 
> > 
> > @asm128, thanks for the advice. Unfortunately that is not an option since the purpose of my code is to integrate with the API for the "third party stuff".
> 
> You can always upgrade what's outdated to the latest version of everyhing and implement what's missing if there's something that got too old or is too difficult to maintain. Why keeping supporting code that doesn't work on the latest release of the language or whatever? I brought to date large C++ projects that were written for Visual C++ 6 without any real difficulties (just they didn't even had to declare the type of the varirable and it assumed int for unknown vars.

@asm128 Sorry, but I think you misunderstand me. I am using an a Python SDK to access a third party API which is incredibly large and maintained by a separate organization. It is far out of scope of my project, and my currently employed role to be honest, to rewrite an entire API SDK just because it relies on libraries that are not yet updated for the latest Python release. 

Additionally, this thread is now very off topic. The point was, installing the latest Visual Studio release should not have prevented me from using one version lower than the latest Python release. Nor should it have removed my conda environment or crashed when attempting to select an environment. I appreciate your willingness to provide feedback, but I don't feel it applies to my current situation or this thread.

**Comment by Roberto-Yanes** [comment]
Hi EveryOne i have the same issue with the Community and Professional versions of VS2022, the Select environment dropdown does not showing the environments, only show the latests Python version installed, and when you try to select after create a new one, the window crashes.  I think that happened when i use a python 3.10 version that is not supported yet. but i uninstall it and install python 3.9 and still crashing.   I tested the same project on vs2019 and works fine.  Do you know how we can solve it? i need to add this project to a .NET 6 main app (i use a python API,  due the numpy library functions that i can't port)  and i can´t do it because VS2019 doesn't support Net 6. 

Thanks in advance. 
Greetings

**Comment by asm128** [comment]
I find this insulting. Fix the product if you want me to open my project in VS2022.

![image](https://user-images.githubusercontent.com/26621255/164369331-27eaf127-7ff7-4fac-a9a8-f25ea08b536c.png)

**Comment by Roberto-Yanes** [comment]
I solve my problem uninstalling all visual studio versions and all python versions. 
After that,  install python from windows store. It crashes if you install the python.org installer.
After i install the VS 2022 pro version and it worked correctly.

**Comment by SteveM49** [comment]
I too tried to switch conda envs and had VS 17.1.5 crash repeatedly.  Roberto-Yanes says he uninstalled all VS versions and all python versions, then reinstalled python from windows store.  I am reluctant to do that since I already have envs set up that support existing systems.  I don't want to lose them.  Is there another solution?  Are the developers of VS looking at this?  Has a ticket been created?

**Comment by alphamonda** [comment]
I also have this issue. With only the Microsoft version Python 3.9 it worked, but when I added Annaconda the issues started. As soon as I try to open the packaging manager Visual Studio crashes. I also recognized that it does not support Python 3.10 which I tried to install at some point. 

Currently I use: Microsoft Visual Studio Community 2022 (64-bit) - Current Version 17.1.6

**Comment by heathermrauch** [comment]
> @heathermrauch, Sorry for the issues you are seeing. Unfortunately, we have been unable to reproduce this. Could you please try installing the latest Python from python.org. There have been some issues with `pip` and want to eliminate that possibility.

@judej I had to set up a new pc from scratch, so I figured I'd give this another go. I was finally able to get this to run with python 3.9, but only after using a brand new computer without anything but anaconda installed. Something from previously installed apps or settings appears to be conflicting with the build. 

Now that I have this installed and "working", I have another question, how do I change the keyboard shortcut for "Send to python interactive" back to 'ctrl + Enter'? Having to hit 'Ctrl + E, Ctrl + E' is really dumb (yes you do have to do it twice like an idiot). I tried changing the keyboard shortcut bindings in the Options menu, which did absolutely nothing. 
![image](https://user-images.githubusercontent.com/81587916/168373979-3f627da0-5272-4090-a9fc-a3afd8efd2fc.png)
Please provide explicit instructions on how to change this.

I also wanted to note, as I was clicking through the various items listed in the options, the page does not change until you click an item, click something else, then click the item you want a second time.

**Comment by StellaHuang95** [comment]
Hi @Roberto-Yanes @alphamonda @SteveM49, sorry for the issues you are experiencing. We have been trying to reproduce this but were not able to. Could you please share your repro steps from a clean install? That would be really helpful. :blush: Thanks!

**Comment by alphamonda** [comment]
Hi, I do not recall the precise steps, but the issue occurred when: 

*	I first had some normal Python version installed
*	I then installed the newest which was 3.10
*	Then I installed Python via Anaconda, from there the crashes started. 

I think it has to do with the newest version of Python which was not “known” to Visual Studio.

 �

Now I deinstalled all Python versions. I only use Anacona with Python 3.9. Since then it does not crash any more. 

Hope that helps a bit. 

Best regards, 

Oliver

 �

**Comment by bschnurr** [comment]
the first two versions of python did they come from the Visual Studio installer? or windows store? or direct download from the python site?

**Comment by judej** [comment]
Closing old issue. If this is still a problem, please reopen with the information requested. thanks

**Comment by alphamonda** [comment]
The first ones were directly installed from the python site. 

 �
